modules/transformer-kinesis/src/main/scala/com/snowplowanalytics/snowplow/rdbloader/transformer/kinesis/Main.scala

```diff
@@ -32,10 +35,24 @@ object Main extends IOApp {
       parsed <- ShredderCliConfig.Stream.loadConfigFrom[IO](BuildInfo.name, BuildInfo.description)(args: Seq[String]).value
       res <- parsed match {
         case Right(cliConfig) =>
-          Resources.mk[IO](cliConfig.igluConfig, cliConfig.config.queue, cliConfig.config.monitoring.metrics).use { resources =>
+          Resources.mk[IO](
+            cliConfig.igluConfig,
+            cliConfig.config.queue,
+            cliConfig.config.monitoring.metrics,
+            executionContext
+          ).use { resources =>
+            val telemetry =
+              Telemetry.run[IO](
```

Comment:
I feel like it could be also part of initialisation of whole app so somewhere in resources? Maybe we could split the `run` method to sth like `make` and `run` so the `make` (returning different impls of telemetry based on enabled/disabled config) is invoked when we build resources and then we `run` when merging streams?